sympy/physics/mechanics/joint.py

```diff
@@ -1046,3 +1079,302 @@ def _set_linear_velocity(self):
         self.child_point.set_vel(self.child.frame, 0)
         self.child_point.set_vel(self.parent.frame, self.speeds[0] * axis)
         self.child.masscenter.set_vel(self.parent.frame, self.speeds[0] * axis)
+
+
+class CylindricalJoint(_JointAxisMixin, Joint):
+    """Cylindrical Joint.
+
+    .. image:: CylindricalJoint.svg
+        :align: center
+        :width: 600
+
+    Explanation
+    ===========
+
+    A cylindrical joint is defined such that the child body rotates about as
+    well as translates along the body fixed joint axis with respect to the
+    parent body. So this joint axis is the the rotation axis as well as the
+    translation axis. The location of the joint is defined by two points, one in
+    each body, which coincide when the generalized coordinate corresponding to
+    the translation is zero. The direction cosine matrix between the
+    childinterframe and parentinterframe is formed using a simple rotation about
```

Comment:
Are "childinterframe" and "parentinterframe" consistently referred with this stylising throughout the docs? Should "child interframe" and "parent interframe" be preferred? I think the later is more readable.

Reply:
yeah should be the later. The full formulation would be intermediate frame of the child body. A bit shorter would be child intermediate frame, of which child interframe is kind of the abbreviation. The attribute is also named ``child_interframe``. So I think it is best to indeed use the later "child interframe".